### ISSUE TITLE
Add current job ID and current cluster name to banyanconfig.toml

### ISSUE THREAD

**Issue by calebwin** [banyan-team/banyan-julia#63]
Currently we have a set of configuration parameters which are stored in `banyanconfig.toml`, environment variables, or parameters to the `configure` function.

- [ ] Make `configure` also store and maintain cluster name and job ID
- [ ] Migrate all functions from using `cluster_name`/`name` as a parameter to simply calling `configure`
- [ ] Migrate all functions from using `current_job_id` to simply calling `configure`

This will improve the consistency across the API and make it super easy for a user to understand and specify what Banyan account, AWS account, cluster, and job are being used at any point.

**Comment by cailinw** [comment]
Won't this make it hard to run multiple jobs at the same time? e.g., running multiple jobs from different processes

**Comment by calebwin** [comment]
Good point, I hadn't thought about that. I think we should still do this but the only change should be that we should cache all configuration parameters in memory. So for each configuration parameter, we should have a global variable to cache that parameters.
